polkadot/runtime/parachains/src/scheduler.rs

```diff
@@ -573,26 +604,32 @@ impl<T: Config> Pallet<T> {
 		if ValidatorGroups::<T>::decode_len().map_or(true, |l| l == 0) {
 			return
 		}
-		// If there exists a core, ensure we schedule at least one job onto it.
-		let n_lookahead = Self::claimqueue_lookahead().max(1);
 		let n_session_cores = T::AssignmentProvider::session_core_count();
 		let cq = ClaimQueue::<T>::get();
 		let config = configuration::ActiveConfig::<T>::get();
+		// Extra sanity, config should already never be smaller than 1:
+		let n_lookahead = config.scheduler_params.lookahead.max(1);
 		let max_availability_timeouts = config.scheduler_params.max_availability_timeouts;
 		let ttl = config.scheduler_params.ttl;
 
 		for core_idx in 0..n_session_cores {
 			let core_idx = CoreIndex::from(core_idx);
 
+			let n_lookahead_used = cq.get(&core_idx).map_or(0, |v| v.len() as u32);
+
 			// add previously timedout paras back into the queue
 			if let Some(mut entry) = timedout_paras.remove(&core_idx) {
 				if entry.availability_timeouts < max_availability_timeouts {
 					// Increment the timeout counter.
 					entry.availability_timeouts += 1;
-					// Reset the ttl so that a timed out assignment.
-					entry.ttl = now + ttl;
-					Self::add_to_claimqueue(core_idx, entry);
-					// The claim has been added back into the claimqueue.
+					if n_lookahead_used < n_lookahead {
+						entry.ttl = now + ttl;
+					} else {
+						// Over max capacity, we need to bump ttl:
```

Comment:
why? is it because it'll take an extra block for the assignment to become part of the lookahead window?

Reply:
Exactly.